sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -225,6 +244,14 @@ export class RuleResourceSerializer implements AtomXmlSerializer {
   }
 }
 
+/**
+ * @internal
+ * @ignore
+ */
+export function isSqlRuleAction(action: any): action is SqlRuleAction {
+  return action != null && typeof action === "object" && "sqlExpression" in action;
```

Comment:
SqlRuleAction is an alias for SqlRuleFilter
`sqlExpression` in SQLRuleFilter is optional.
So, when user is creating a rule or updating a rule, they can pass {} as filter?

Reply:
```
export interface SqlRuleFilter {
  /**
   * SQL expression to use.
   */
  sqlExpression?: string;

  /**
   * SQL parameters to the expression
   */
  sqlParameters?: SqlParameter[];

  /**
   * This property is reserved for future use. An integer value showing the
   * compatibility level, currently hard-coded to 20.
   */
  compatibilityLevel?: number;

  /**
   * Boolean value indicating whether the SQL filter expression requires preprocessing
   */
  requiresPreprocessing?: boolean;
}
```
I was wondering if I should make this `sqlExpression` required.
I don't know what `compatibilityLevel` is. The other two attributes are meaningless if the `sqlExpression` is undefined.

(That's the main intention behind looking for `sqlExpression` in the `isSqlRuleAction`.)

Should I make the `sqlExpression` a required property?

Reply:
But the `sqlExpression` can be undefined if not set.
Hmmm.. I'll think about it again!

Reply:
Yes, `sqlExpression` should be mandatory in `SqlRuleFilter` and `SqlRuleAction`

Reply:
Service supports "EmptyRuleAction" for the undefined rule action, making the sqlExpression a required attribute would mean that the SDK won't be allowing empty rule action which might become an overhead for the service while filtering the rules.

So, I'd like to keep the sqlExpression as an optional attribute.

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/10049/commits/093fe8e41393c9944334e062dd85021f8885e3e8